packages/core/src/builtin-tasks/node.ts

```diff
@@ -30,6 +41,38 @@ subtask('node:start', 'start a ckb node')
       detached,
       genesisAccountArgs: genesisArgs,
     })
+
+    env.config.network = ckbDockerNetwork.url
+
+    const builtInDirPath = cachePath('built-in')
+    for (const script of BUILTIN_SCRIPTS) {
+      const filePath = path.join(builtInDirPath, script)
+      if (!fs.existsSync(filePath)) {
+        const out = fs.createWriteStream(path.resolve(builtInDirPath, script))
+        await undici.stream(
+          `${
+            env.config.devNode?.builtInContractDownloadBaseUrl ?? DEFAULT_BUILTIN_CONTRACT_DOWNLOAD_BASE_URL
+          }/${script}}`,
+          {
+            opaque: out,
+            method: 'GET',
+          },
+          ({ opaque }) => opaque as fs.WriteStream,
+        )
+      }
+    }
+    await scheduler.wait(20000)
+    await ckbDockerNetwork.generateLumosConfig()
+    config.initializeConfig(ckbDockerNetwork.lumosConfig)
+
+    await ckbDockerNetwork.deployScripts({
+      builtInScriptName: BUILTIN_SCRIPTS,
+      configFilePath: env.config.devNode?.builtInContractConfigPath ?? path.resolve(configPath(), 'scripts.json'),
+      builtInDirPath,
+      indexer: new Indexer(ckbDockerNetwork.url),
+      rpc: new RPC(ckbDockerNetwork.url),
+      privateKey: env.config.kuaiArguments.privateKey ?? DEFAULT_KUAI_PRIVATE_KEY,
+    })
```

Comment:
I was thinking that the deployment of these contracts should be a function maintained by the `ckb node` itself, and that different `ckb nodes` might have different ways of deploying built-in contracts, similar to the way `genesisAccountArgs` is passed above

BTW, maybe we should abstract `ckb node` into an interface, so that if we want to start `ckb node` in other ways, we can just implement the corresponding interface

Reply:
>  that different `ckb nodes` might have different ways of deploying built-in contracts

Could you elaborate on the reason that different node types will have different progress to deploy built-in contracts?

In my mind, the progress can be abstracted into the following steps:

1. build contract source code into binaries
2. send the binaries one by one to the target node

There's no keyword of node type bifurcates the workflow.

> BTW, maybe we should abstract ckb node into an interface, so that if we want to start ckb node in other ways, we can just implement the corresponding interface

+1

Reply:
> I was thinking that the deployment of these contracts should be a function maintained by the `ckb node` itself, and that different `ckb nodes` might have different ways of deploying built-in contracts, similar to the way `genesisAccountArgs` is passed above
> 
> BTW, maybe we should abstract `ckb node` into an interface, so that if we want to start `ckb node` in other ways, we can just implement the corresponding interface

Yes, I think all these could be done by abstracting an interface. The `ckb docker node` is an implementation of the interface.

Reply:
I think we can do this in another branch, which is specifically for this.

Reply:
Please open a new issue for track